src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,82 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();

```

Comment:
The `RegistrationServiceImpl` instance is shared across all tests. This could lead to interference between tests if the `StorageDao` is not reset between tests. Consider using a fresh instance of `RegistrationServiceImpl` for each test or resetting the storage state before each test to ensure test isolation.